src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -0,0 +1,455 @@
+  rw [@nat_degree_X_pow_sub_C K _ _ n hpos 1, ← nth_roots],
+  exact is_primitive_root.card_nth_roots h
+end
+
+/-- `cyclotomic' n K` splits. -/
+lemma cycl_splits (n : ℕ) : splits (ring_hom.id K) (cyclotomic' n K) :=
+begin
+  apply splits_prod (ring_hom.id K),
+  intros z hz,
+  simp only [splits_X_sub_C (ring_hom.id K)]
+end
+
+/-- If there is a primitive `n`-th root of unity in `K`, then `X ^ n - 1`splits. -/
+lemma X_pow_sub_one_splits {ζ : K} {n : ℕ} (h : is_primitive_root ζ n) :
+  splits (ring_hom.id K) (X ^ n - C (1 : K)) :=
+begin
+  by_cases hzero : n = 0,
+  { simp only [hzero, ring_hom.map_one, splits_zero, pow_zero, sub_self] },
+  apply (splits_iff_card_roots (X_pow_sub_C_ne_zero (nat.pos_of_ne_zero hzero) (1 : K))).2,
+  rw [← nth_roots, is_primitive_root.card_nth_roots h, nat_degree_X_pow_sub_C],
+  exact nat.pos_of_ne_zero hzero
+end
+
+/-- If there is a primitive `n`-th root of unity in `K`, then
+`X ^ n - 1 = ∏ (cyclotomic' i K)`, where `i` divides `n`. -/
+lemma X_pow_sub_one_eq_prod_cycl' {ζ : K} {n : ℕ} (hpos : 0 < n) (h : is_primitive_root ζ n) :
+  X ^ n - 1 = ∏ i in nat.divisors n, cyclotomic' i K :=
+begin
+  rw [X_pow_sub_one_eq_prod hpos h],
+  have rwcyc : ∀ i ∈ nat.divisors n, cyclotomic' i K = ∏ μ in primitive_roots i K, (X - C μ),
+  { intros i hi,
+    simp only [cyclotomic'] },
+  conv_rhs { apply_congr,
+             skip,
+             simp [rwcyc, H] },
+  rw ← finset.prod_bind,
+  { simp only [is_primitive_root.nth_roots_one_eq_bind_primitive_roots hpos h] },
+  intros x hx y hy hdiff,
+  simp only [nat.mem_divisors, and_true, ne.def, pnat.ne_zero, not_false_iff] at hx hy,
+  refine is_primitive_root.disjoint _ _ hdiff,
+  { exact @nat.pos_of_mem_divisors n x (nat.mem_divisors.2 hx) },
+  { exact @nat.pos_of_mem_divisors n y (nat.mem_divisors.2 hy) }
+end
+
+/-- If there is a primitive `n`-th root of unity in `K`, then
+`cyclotomic' n K = (X ^ k - 1) /ₘ (∏ i in nat.proper_divisors k, cyclotomic' i K)`. -/
+lemma cyclotomic'_eq_X_pow_sub_one_div {ζ : K} {n : ℕ} (hpos: 0 < n) (h : is_primitive_root ζ n) :
+  cyclotomic' n K = (X ^ n - 1) /ₘ (∏ i in nat.proper_divisors n, cyclotomic' i K) :=
+begin
+  rw [X_pow_sub_one_eq_prod_cycl' hpos h,
+  nat.divisors_eq_proper_divisors_insert_self_of_pos hpos,
+  finset.prod_insert nat.proper_divisors.not_self_mem],
+  have prod_monic : (∏ i in nat.proper_divisors n, cyclotomic' i K).monic,
+  { apply monic_prod_of_monic,
+    intros i hi,
+    exact cyclotomic'.monic i K },
+  rw (div_mod_by_monic_unique (cyclotomic' n K) 0 prod_monic _).1,
+  simp only [degree_zero, zero_add],
+  split,
+  { rw mul_comm },
+  rw [bot_lt_iff_ne_bot],
+  intro h,
+  exact monic.ne_zero prod_monic (degree_eq_bot.1 h)
+end
+
+/-- If there is a primitive `n`-th root of unity in `K`, then `cyclotomic' n K` comes from a
+polynomial with integer coefficients. -/
+lemma int_coeff_of_cycl {ζ : K} {n : ℕ} (h : is_primitive_root ζ n) :
```

Comment:
Should this be rephrased in terms of `lifts`?

Reply:
It uses `lifts` in the proof, but the statement is more precise (the lifting polynomial is monic and of the correct degree). I proved in `lifts` that the two notions are the same (this is the first line of the proof), but the only definition is `lifts`, the subring of polynomial that lifts, so at some point I have to write by hand a lemma with the other properties.

Reply:
Aah, you are right.